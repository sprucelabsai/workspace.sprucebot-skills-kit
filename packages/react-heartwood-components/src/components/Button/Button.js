// @flow
import React from 'react'
import cx from 'classnames'
import is from 'is_js'
import Loader from '../Loader/Loader'
import Icon from '../Icon/Icon'
import BasicAnchor from '../_utilities/Anchor'

import type { Node, Element } from 'react'
import type { Props as IconProps } from '../Icon/Icon'

export type Props = {
	/** Optional class to add to the button. */
	className?: string,

	/** Sets the visual appearance of the button. May be primary, secondary, simple, or caution. */
	kind?: string,

	/** Set true to make the button less tall. */
	isSmall?: boolean,

	/** Set true to make the button fill its parent's width. */
	isFullWidth?: boolean,

	/** Set true to hide any text or icon in the button and show a loader instead. */
	isLoading?: boolean,

	/** Set true to hide any text in the button. Text should still be provided for accessibility. */
	isIconOnly?: boolean,

	/** Text for the button. */
	text?: string,

	/** Will render a link. May be relative or absolute. */
	href?: string,

	/** Icon for the button. */
	icon?: Node | IconProps,

	/** Type attribute for HTML button element. Defaults to 'button'. */
	type?: string,

	/** Click handler. */
	onClick?: Function,

	/** Will be passed back with the on click. */
	payload?: Object,

	/** Component used to render anchor */
	AnchorComponent?: Node
}

const Button = (props: Props) => {
	const {
		className,
		kind,
		isSmall,
		isFullWidth,
		isLoading,
		isIconOnly,
		text,
		href,
		icon,
		type,
		onClick,
		payload,
		AnchorComponent = BasicAnchor,
		...rest
	} = props
	const btnClass = cx(className, {
		btn: true,
		'btn-primary': kind === 'primary',
		'btn-secondary': kind === 'secondary',
		'btn-caution': kind === 'caution',
		'btn-simple': kind === 'simple',
		'btn-full-width': isFullWidth,
		'btn--loading': isLoading,
		'btn-small': isSmall,
		'btn-icon-only': !text || isIconOnly
	})
	const textClass = cx('btn__text', {
		'visually-hidden': isIconOnly
	})

	const handleClick = (e: any) => {
		e.currentTarget.blur()
		if (onClick) {
			onClick(e, props.payload)
		}
	}

	const Inner = () => (
		<span className="btn__inner">
			{icon && (icon.customIcon || icon.name) && (
				<span className="btn__icon-wrapper">
					<Icon
						customIcon={icon.customIcon}
						icon={icon.name}
						isLineIcon={icon.isLineIcon}
						className={cx(
							{
								btn__icon: true,
								'btn__line-icon': icon.isLineIcon
							},
							icon.className
						)}
					/>
				</span>
			)}
			{text && <span className={textClass}>{text}</span>}
			{isLoading && <Loader />}
		</span>
	)

	const button = (
		<button className={btnClass} type={type} onClick={handleClick} {...rest}>
			<Inner />
		</button>
	)

	const anchor = (
		<AnchorComponent href={href} className={btnClass} {...rest}>
			<Inner />
		</AnchorComponent>
	)

	if (!text && !icon) {
		// TODO: Handle Logging
		// console.error(
		// 	'<Button /> must have text, icon, or both. Please check the props your passing.'
		// )
		return null
	}

	return href ? anchor : button
}

Button.defaultProps = {
	className: '',
	kind: '',
	isSmall: false,
	isFullWidth: false,
	isLoading: false,
	isIconOnly: false,
	text: '',
	href: '',
	icon: null,
	type: 'button',
	onClick: () => null
}

export default Button
