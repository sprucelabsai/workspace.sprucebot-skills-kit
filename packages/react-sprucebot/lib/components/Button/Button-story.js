"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _react2 = require("@storybook/react");

var _react3 = require("@storybook/addon-knobs/react");

var _Wrapper = _interopRequireDefault(require("../../../.storybook/Wrapper"));

var _Button = _interopRequireDefault(require("./Button"));

var Icon11 = function Icon11(props) {
  return _react.default.createElement("svg", props, _react.default.createElement("path", {
    clipRule: "evenodd",
    d: "M4.285 1.49h-.011A4.5 4.5 0 0 0 .75 5.887v.89a1.5 1.5 0 0 0 1.5 1.5H6a1.5 1.5 0 0 0 1.5-1.5 1.5 1.5 0 0 1 1.5-1.5h6a1.5 1.5 0 0 1 1.5 1.5 1.5 1.5 0 0 0 1.5 1.5h3.75a1.5 1.5 0 0 0 1.5-1.5v-.89a4.5 4.5 0 0 0-3.524-4.393h-.011a40.593 40.593 0 0 0-15.43-.002zM12.305 11.275H11.7a9 9 0 0 0-8.931 7.884l-.093.744a3 3 0 0 0 2.977 3.372h12.7a3 3 0 0 0 2.975-3.375l-.093-.744a9 9 0 0 0-8.93-7.88z",
    strokeWidth: "1.5",
    strokeLinecap: "round",
    strokeLinejoin: "round"
  }), _react.default.createElement("path", {
    d: "M9.75 11.487V9.025M14.25 11.487V9.025",
    strokeWidth: "1.5",
    strokeLinecap: "round",
    strokeLinejoin: "round"
  }), _react.default.createElement("path", {
    clipRule: "evenodd",
    d: "M12 20.275a3 3 0 1 0 0-6 3 3 0 0 0 0 6z",
    strokeWidth: "1.5",
    strokeLinecap: "round",
    strokeLinejoin: "round"
  }));
};

Icon11.defaultProps = {
  width: "24",
  height: "25",
  viewBox: "0 0 24 25",
  xmlns: "http://www.w3.org/2000/svg"
};

var ButtonGroup = function ButtonGroup() {
  var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {
    kind: ''
  };
  var btnText = (0, _react3.text)('Text', 'Hello World');
  var isSmall = (0, _react3.boolean)('Small', false);
  var isFullWidth = (0, _react3.boolean)('Full Width', false);
  return _react.default.createElement(_react.Fragment, null, _react.default.createElement(_Button.default, (0, _defineProperty2.default)({
    className: "l-mr-small l-mb-small",
    isSmall: isSmall,
    isFullWidth: isFullWidth,
    text: btnText,
    kind: props.kind || ''
  }, "isFullWidth", false)), _react.default.createElement(_Button.default, {
    className: "l-mr-small l-mb-small",
    isSmall: isSmall,
    isFullWidth: isFullWidth,
    text: "Link ".concat(btnText),
    kind: props.kind || '',
    href: "#",
    target: "_blank"
  }), _react.default.createElement(_Button.default, {
    className: "l-mr-small l-mb-small",
    isSmall: isSmall,
    isFullWidth: isFullWidth,
    text: btnText,
    kind: props.kind || '',
    disabled: true
  }), _react.default.createElement(_Button.default, {
    className: "l-mr-small l-mb-small",
    isSmall: isSmall,
    isFullWidth: isFullWidth,
    text: btnText,
    kind: props.kind || '',
    disabled: true,
    isLoading: true
  }), _react.default.createElement(_Button.default, {
    className: "l-mr-small l-mb-small",
    isSmall: isSmall,
    isFullWidth: isFullWidth,
    text: btnText,
    kind: props.kind || '',
    icon: _react.default.createElement(Icon11, {
      className: "btn__line-icon"
    })
  }), _react.default.createElement(_Button.default, {
    className: "l-mr-small l-mb-small",
    isSmall: isSmall,
    isFullWidth: isFullWidth,
    kind: props.kind || '',
    icon: _react.default.createElement(Icon11, {
      className: "btn__line-icon"
    })
  }));
};

var stories = (0, _react2.storiesOf)('Button', module);
stories.addDecorator(_react3.withKnobs);
stories.add('Primary', function () {
  return _react.default.createElement(ButtonGroup, {
    kind: "primary"
  });
}).add('Secondary', function () {
  return _react.default.createElement(ButtonGroup, {
    kind: "secondary"
  });
}).add('Simple', function () {
  return _react.default.createElement(ButtonGroup, {
    kind: "simple"
  });
}).add('Caution', function () {
  return _react.default.createElement(ButtonGroup, {
    kind: "caution"
  });
});