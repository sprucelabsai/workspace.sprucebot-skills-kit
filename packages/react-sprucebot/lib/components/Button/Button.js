'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Loader = require('../Loader/Loader');

var _Loader2 = _interopRequireDefault(_Loader);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

// TODO refactor into styled component
var Button = function (_Component) {
	_inherits(Button, _Component);

	function Button(props) {
		_classCallCheck(this, Button);

		var _this = _possibleConstructorReturn(this, (Button.__proto__ || Object.getPrototypeOf(Button)).call(this, props));

		_this.onClick = function (e) {
			if (_this.props.onClick) {
				_this.props.onClick(e);
			} else if (_this.props.href) {
				e.preventDefault();
				_this.setState({ busy: true });
				var url = _this.props.href;
				if (/^http/.test(url)) {
					// If the href is a full domain name
					if (_this.props.target) {
						window.open(url, _this.props.target);
					} else {
						window.open(url, '_self');
					}
				} else {
					// Relative url
					if (_this.props.target) {
						window.open(url, _this.props.target);
					} else if (_this.props.router) {
						_this.props.router.push(url);
					} else {
						window.open(url, '_self');
					}
				}

				// Reset the state to not-busy if it's been 2 sec
				setTimeout(function () {
					_this.setState({ busy: false });
				}, 2000);
			}
		};

		_this.state = {
			busy: !!props.busy
		};
		return _this;
	}

	_createClass(Button, [{
		key: 'componentWillReceiveProps',
		value: function componentWillReceiveProps(nextProps) {
			if (typeof nextProps.busy !== 'undefined') {
				this.setState({
					busy: nextProps.busy
				});
			}
		}
	}, {
		key: 'render',
		value: function render() {
			var _props = this.props,
			    tag = _props.tag,
			    disabled = _props.disabled,
			    primary = _props.primary,
			    secondary = _props.secondary,
			    alt = _props.alt,
			    link = _props.link,
			    caution = _props.caution,
			    className = _props.className,
			    children = _props.children,
			    submit = _props.submit,
			    remove = _props.remove,
			    toggle = _props.toggle,
			    router = _props.router,
			    loaderDark = _props.loaderDark,
			    loaderStyle = _props.loaderStyle,
			    propBusy = _props.busy,
			    hideLoader = _props.hideLoader,
			    props = _objectWithoutProperties(_props, ['tag', 'disabled', 'primary', 'secondary', 'alt', 'link', 'caution', 'className', 'children', 'submit', 'remove', 'toggle', 'router', 'loaderDark', 'loaderStyle', 'busy', 'hideLoader']);

			var busy = this.state.busy;


			if (primary && secondary) {
				return _react2.default.createElement(
					'button',
					{ className: 'btn__primary' },
					'\'primary\' and \'secondary\' are mutually exclusive.'
				);
			} else if (primary && alt) {
				return _react2.default.createElement(
					'button',
					{ className: 'btn__primary' },
					'\'primary\' and \'alt\' are mutually exclusive.'
				);
			}

			var btnClass = primary ? 'btn__primary' : '';
			btnClass += secondary ? 'btn__secondary' : '';
			btnClass += alt && btnClass.length > 0 ? '__alt' : '';
			btnClass += alt && btnClass.length === 0 ? 'btn__alt' : '';
			btnClass += disabled ? ' btn__disabled' : '';
			btnClass += caution ? ' btn__caution' : '';
			btnClass += link ? ' btn__link' : '';
			btnClass += toggle ? 'btn__toggle' : '';

			if (remove) {
				btnClass = 'btn__remove';
			} else if (!btnClass) {
				btnClass = 'btn';
			}

			// if this button has a href or is a "remove" button, make it an anchor
			var Tag = props.href || remove ? 'a' : tag;

			return _react2.default.createElement(
				Tag,
				_extends({
					className: btnClass + ' ' + (className || ''),
					onClick: this.onClick
				}, props),
				busy && !hideLoader ? _react2.default.createElement(_Loader2.default, {
					dark: loaderDark ? true : false,
					fullWidth: false,
					loaderStyle: loaderStyle
				}) : children
			);
		}
	}]);

	return Button;
}(_react.Component);

exports.default = Button;


Button.propTypes = {
	tag: _propTypes2.default.string,
	primary: _propTypes2.default.bool,
	alt: _propTypes2.default.bool,
	secondary: _propTypes2.default.bool,
	busy: _propTypes2.default.bool,
	href: _propTypes2.default.string,
	remove: _propTypes2.default.bool,
	toggle: _propTypes2.default.bool,
	hideLoader: _propTypes2.default.bool,
	type: _propTypes2.default.string
};

Button.defaultProps = {
	tag: 'button',
	primary: false,
	alt: false,
	secondary: false,
	busy: false,
	remove: false,
	toggle: false,
	type: 'button'
};