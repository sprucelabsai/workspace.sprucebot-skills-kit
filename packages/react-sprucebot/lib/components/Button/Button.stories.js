'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _react3 = require('@storybook/react');

var _addonActions = require('@storybook/addon-actions');

var _addonInfo = require('@storybook/addon-info');

var _react4 = require('@storybook/addon-knobs/react');

var _storybookReadme = require('storybook-readme');

var _Button = require('./Button');

var _Button2 = _interopRequireDefault(_Button);

var _Button3 = require('./Button.md');

var _Button4 = _interopRequireDefault(_Button3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var stories = (0, _react3.storiesOf)('Button', module);
stories.addDecorator(_react4.withKnobs);

stories.add('Primary', (0, _storybookReadme.withReadme)(_Button4.default, (0, _addonInfo.withInfo)()(function () {
	return _react2.default.createElement(
		_Button2.default,
		{ primary: true, onClick: (0, _addonActions.action)('Button onClick') },
		(0, _react4.text)('Button Text', 'Primary Button ðŸ˜€ðŸŒ²ðŸ¤–')
	);
}))).add('Secondary', (0, _storybookReadme.withReadme)(_Button4.default, (0, _addonInfo.withInfo)()(function () {
	return _react2.default.createElement(
		_Button2.default,
		{ secondary: true, onClick: (0, _addonActions.action)('Button onClick') },
		(0, _react4.text)('Button Text', 'Secondary Button')
	);
}))).add('Alt', (0, _storybookReadme.withReadme)(_Button4.default, (0, _addonInfo.withInfo)()(function () {
	return _react2.default.createElement(
		_Button2.default,
		{ alt: true, onClick: (0, _addonActions.action)('Button onClick') },
		(0, _react4.text)('Button Text', 'Alt Button')
	);
}))).add('Disabled', (0, _storybookReadme.withReadme)(_Button4.default, (0, _addonInfo.withInfo)()(function () {
	return _react2.default.createElement(
		'div',
		null,
		_react2.default.createElement(
			_Button2.default,
			{ primary: true, disabled: true, onClick: (0, _addonActions.action)('Button onClick') },
			(0, _react4.text)('Button Text', 'Primary Button Disabled')
		),
		_react2.default.createElement(
			_Button2.default,
			{ alt: true, disabled: true, onClick: (0, _addonActions.action)('Button onClick') },
			(0, _react4.text)('Button Text', 'Secondary Button Disabled')
		)
	);
}))).add('Caution', (0, _storybookReadme.withReadme)(_Button4.default, (0, _addonInfo.withInfo)()(function () {
	return _react2.default.createElement(
		_Button2.default,
		{ primary: true, caution: true, onClick: (0, _addonActions.action)('Button onClick') },
		(0, _react4.text)('Button Text', 'Take Me To The Danger Zone')
	);
}))).add('Tertiary', (0, _storybookReadme.withReadme)(_Button4.default, (0, _addonInfo.withInfo)()(function () {
	return _react2.default.createElement(
		_Button2.default,
		{ tertiary: true, onClick: (0, _addonActions.action)('Button onClick') },
		(0, _react4.text)('Button Text', 'Minor Button')
	);
}))).add('Loading', (0, _storybookReadme.withReadme)(_Button4.default, (0, _addonInfo.withInfo)()(function () {
	return _react2.default.createElement(
		'div',
		null,
		_react2.default.createElement(
			_Button2.default,
			{ primary: true, busy: true, onClick: (0, _addonActions.action)('Button onClick') },
			(0, _react4.text)('Button Text', 'Primary Button Disabled')
		),
		_react2.default.createElement(
			_Button2.default,
			{ alt: true, busy: true, onClick: (0, _addonActions.action)('Button onClick') },
			(0, _react4.text)('Button Text', 'Secondary Button Disabled')
		)
	);
}))).add('Toggle', (0, _storybookReadme.withReadme)(_Button4.default, (0, _addonInfo.withInfo)()(function () {
	return _react2.default.createElement(
		_Button2.default,
		{ toggle: true, onClick: (0, _addonActions.action)('Button onClick') },
		(0, _react4.text)('Button Text', 'Toggle Button')
	);
}))).add('Remove', (0, _storybookReadme.withReadme)(_Button4.default, (0, _addonInfo.withInfo)()(function () {
	return _react2.default.createElement(
		_Button2.default,
		{ primary: true, remove: true, onClick: (0, _addonActions.action)('Button onClick') },
		(0, _react4.text)('Button Text', 'Toggle Button')
	);
})));