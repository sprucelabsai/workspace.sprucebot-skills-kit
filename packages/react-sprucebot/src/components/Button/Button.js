import React, { Component } from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import Loader from '../Loader/Loader'
import Link from 'next/link'

const ButtonWrapper = styled.div`
	display: flex;
	width: 50%;
	${props => props.left && `padding-right: 1.125em;`};
	${props => props.right && `padding-left: 1.125em;`};
`

const StyledButton = styled.button`
	${props =>
		props.busy ||
		(props.disabled &&
			`
			pointer-events: none;
			cursor: not-allowed;
		`)};
`

const StyledAnchor = styled.a`
	${props =>
		props.busy ||
		(props.disabled &&
			`
		pointer-events: none;
		cursor: not-allowed;
	`)};
`

// TODO refactor into styled component
export default class Button extends Component {
	constructor(props) {
		super(props)
		this.state = {
			busy: !!props.busy
		}
	}

	componentWillReceiveProps(nextProps) {
		if (typeof nextProps.busy !== 'undefined') {
			this.setState({
				busy: nextProps.busy
			})
		}
	}

	onClick = e => {
		const { busy } = this.state
		const { disabled, onClick, href, target, router } = this.props

		if (busy || disabled) {
			return
		}

		if (onClick) {
			onClick(e)
		} else if (href) {
			this.setState({ busy: true })
		}
	}

	renderView = () => {
		const { busy } = this.state
		const { hideLoader, loaderDark, loaderStyle, children } = this.props

		if (busy && !hideLoader) {
			return (
				<Loader
					dark={loaderDark ? true : false}
					fullWidth={false}
					loaderStyle={loaderStyle}
				/>
			)
		}
		return children
	}

	render() {
		const {
			tag,
			disabled,
			primary,
			secondary,
			alt,
			link,
			caution,
			className,
			children,
			submit,
			remove,
			toggle,
			router,
			loaderDark,
			loaderStyle,
			busy: propBusy,
			hideLoader,
			tertiary,
			left,
			right,
			...props
		} = this.props

		const { busy } = this.state

		if (primary && secondary) {
			return (
				<button className="btn__primary">
					'primary' and 'secondary' are mutually exclusive.
				</button>
			)
		} else if (primary && alt) {
			return (
				<button className="btn__primary">
					'primary' and 'alt' are mutually exclusive.
				</button>
			)
		}

		let btnClass = primary ? 'btn__primary' : ''
		btnClass += secondary ? 'btn__secondary' : ''
		btnClass += alt && btnClass.length > 0 ? '__alt' : ''
		btnClass += alt && btnClass.length === 0 ? 'btn__alt' : ''
		btnClass += disabled ? ' btn__disabled' : ''
		btnClass += caution ? ' btn__caution' : ''
		btnClass += link ? ' btn__link' : ''
		btnClass += toggle ? 'btn__toggle' : ''

		if (remove) {
			btnClass = 'btn__remove'
		}

		// if this button has a href or is a "remove" button, make it an anchor
		let Tag
		if (props.href || remove) {
			Tag = Link
		} else if (tag === 'button') {
			Tag = StyledButton
		} else {
			Tag = tag
		}

		if (tertiary) {
			return (
				<ButtonWrapper left={left} right={right}>
					<Tag
						className={`btn ${btnClass} ${className || ''}`}
						onClick={this.onClick}
						disabled={disabled}
						busy={busy}
						{...props}
					>
						<span className="wrapper">{this.renderView()}</span>
					</Tag>
				</ButtonWrapper>
			)
		}

		return (
			<Tag
				className={`btn ${btnClass} ${className || ''}`}
				onClick={this.onClick}
				disabled={disabled}
				busy={busy}
				{...props}
			>
				<span className="wrapper">{this.renderView()}</span>
			</Tag>
		)
	}
}

Button.propTypes = {
	tag: PropTypes.string,
	primary: PropTypes.bool,
	alt: PropTypes.bool,
	secondary: PropTypes.bool,
	busy: PropTypes.bool,
	href: PropTypes.string,
	remove: PropTypes.bool,
	toggle: PropTypes.bool,
	hideLoader: PropTypes.bool,
	tertiary: PropTypes.bool,
	left: PropTypes.bool,
	right: PropTypes.bool,
	type: PropTypes.string
}

Button.defaultProps = {
	tag: 'button',
	primary: false,
	alt: false,
	secondary: false,
	busy: false,
	remove: false,
	toggle: false,
	type: 'button'
}
