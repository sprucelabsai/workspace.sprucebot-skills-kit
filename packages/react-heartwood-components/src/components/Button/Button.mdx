---
name: Button
---

import Button from './Button'

# Button

<Button
	kind="primary"
	className="global_btn__rvopf"
	text="Click it or lose it"
/>
