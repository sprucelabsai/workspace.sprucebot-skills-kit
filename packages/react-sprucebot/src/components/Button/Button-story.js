// @flow
import React, { Fragment } from 'react'
import { storiesOf } from '@storybook/react'
import {
	withKnobs,
	withKnobsOptions,
	text,
	boolean,
	object
} from '@storybook/addon-knobs/react'
import Wrapper from '../../../.storybook/Wrapper'
import Button from './Button'
import Icon from '../Icon/Icon'
import Icon11 from '../../../static/assets/icons/Phones-Mobile-Devices/Phone/phone-retro-1.svg'

type GroupProps = {
	kind: string
}

const btnText = text('text', 'Hello World')
const isSmall = boolean('isSmall', false)
const isFullWidth = boolean('isFullWidth', false)
const icon = text('icon', 'edit')

const stories = storiesOf('Button', module)
stories.addDecorator(
	withKnobsOptions({
		escapeHTML: false
	})
)
stories.addDecorator(withKnobs)

stories
	.add('Primary', () => (
		<Button
			className={text('className', 'l-mr-small l-mb-small')}
			isSmall={isSmall}
			isFullWidth={isFullWidth}
			text={btnText}
			kind={text('kind', '') || 'primary'}
			disabled={boolean('disabled', false)}
			isLoading={boolean('isLoading', false)}
			icon={
				icon ? (
					<Icon
						icon={text('icon', '')}
						className={text('iconClassName', 'btn__line-icon')}
					/>
				) : null
			}
			href={text('href', '')}
			target={text('target', '')}
			onClick={text('onClick', '() => console.log("you clicked")')}
			linkProps={object('linkProps', {})}
		/>
	))
	.add('Secondary', () => (
		<Button
			className={text('className', 'l-mr-small l-mb-small')}
			isSmall={isSmall}
			isFullWidth={isFullWidth}
			text={btnText}
			kind={text('kind', '') || 'secondary'}
			disabled={boolean('disabled', false)}
			isLoading={boolean('isLoading', false)}
			icon={
				icon ? (
					<Icon
						icon={text('icon', '')}
						className={text('iconClassName', 'btn__line-icon')}
					/>
				) : null
			}
			href={text('href', '')}
			target={text('target', '')}
			onClick={text('onClick', '() => console.log("you clicked")')}
			linkProps={object('linkProps', {})}
		/>
	))
	.add('Simple', () => (
		<Button
			className={text('className', 'l-mr-small l-mb-small')}
			isSmall={isSmall}
			isFullWidth={isFullWidth}
			text={btnText}
			kind={text('kind', '') || 'simple'}
			disabled={boolean('disabled', false)}
			isLoading={boolean('isLoading', false)}
			icon={
				icon ? (
					<Icon
						icon={text('icon', '')}
						className={text('iconClassName', 'btn__line-icon')}
					/>
				) : null
			}
			href={text('href', '')}
			target={text('target', '')}
			onClick={text('onClick', '() => console.log("you clicked")')}
			linkProps={object('linkProps', {})}
		/>
	))
	.add('Caution', () => (
		<Button
			className={text('className', 'l-mr-small l-mb-small')}
			isSmall={isSmall}
			isFullWidth={isFullWidth}
			text={btnText}
			kind={text('kind', '') || 'caution'}
			disabled={boolean('disabled', false)}
			isLoading={boolean('isLoading', false)}
			icon={
				icon ? (
					<Icon
						icon={text('icon', '')}
						className={text('iconClassName', 'btn__line-icon')}
					/>
				) : null
			}
			href={text('href', '')}
			target={text('target', '')}
			onClick={text('onClick', '() => console.log("you clicked")')}
			linkProps={object('linkProps', {})}
		/>
	))
