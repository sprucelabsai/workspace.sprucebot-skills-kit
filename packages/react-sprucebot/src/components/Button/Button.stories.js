import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { withInfo } from '@storybook/addon-info'
import { withKnobs, text, boolean, number } from '@storybook/addon-knobs/react'
import { withReadme, withDocs } from 'storybook-readme'
import Button from './Button'
import readme from './Button.md'

const stories = storiesOf('Button', module)

stories.addDecorator(story => (
	<div style={{ width: '500px', margin: 'auto' }}>{story()}</div>
))
stories.addDecorator(withKnobs)

stories
	.add(
		'Primary',
		withReadme(
			readme,
			withInfo()(() => (
				<Button
					primary
					disabled={boolean('Disabled', false)}
					busy={boolean('Busy', false)}
					onClick={action('Button onClick')}
				>
					{text('Button Text', 'Primary Button ðŸ˜€ðŸŒ²ðŸ¤–')}
				</Button>
			))
		)
	)
	.add(
		'Secondary',
		withReadme(
			readme,
			withInfo()(() => (
				<Button
					secondary
					disabled={boolean('Disabled', false)}
					busy={boolean('Busy', false)}
					onClick={action('Button onClick')}
				>
					{text('Button Text', 'Secondary Button')}
				</Button>
			))
		)
	)
	.add(
		'Alt',
		withReadme(
			readme,
			withInfo()(() => (
				<Button
					alt
					disabled={boolean('Disabled', false)}
					busy={boolean('Busy', false)}
					onClick={action('Button onClick')}
				>
					{text('Button Text', 'Alt Button')}
				</Button>
			))
		)
	)
	.add(
		'Caution',
		withReadme(
			readme,
			withInfo()(() => (
				<Button
					primary
					caution
					disabled={boolean('Disabled', false)}
					busy={boolean('Busy', false)}
					onClick={action('Button onClick')}
				>
					{text('Button Text', 'Take Me To The Danger Zone')}
				</Button>
			))
		)
	)
	.add(
		'Toggle',
		withReadme(
			readme,
			withInfo()(() => (
				<Button toggle onClick={action('Button onClick')}>
					{text('Button Text', 'Toggle Button')}
				</Button>
			))
		)
	)
	.add(
		'Remove',
		withReadme(
			readme,
			withInfo()(() => (
				<Button
					primary
					remove
					disabled={boolean('Disabled', false)}
					busy={boolean('Busy', false)}
					onClick={action('Button onClick')}
				>
					{text('Button Text', 'Toggle Button')}
				</Button>
			))
		)
	)
