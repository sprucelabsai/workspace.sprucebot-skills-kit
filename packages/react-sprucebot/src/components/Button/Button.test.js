import React from 'react'
import Button from './Button'
import { shallow } from 'enzyme'

describe('Button tests', () => {
	it('Should match the snapshot', () => {
		const renderedComponent = shallow(<Button />)

		expect(renderedComponent).toMatchSnapshot()
	})
})
