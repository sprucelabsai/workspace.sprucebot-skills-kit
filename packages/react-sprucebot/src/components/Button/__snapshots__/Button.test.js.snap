// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component it renders 1`] = `
<button
  className="btn "
  type="button"
/>
`;

exports[`Button Component it renders alt 1`] = `
<button
  className="btn__alt "
  type="button"
/>
`;

exports[`Button Component it renders busy 1`] = `
<button
  className="btn "
  type="button"
>
  <span
    className="loader_wrapper"
    style={Object {}}
  >
    <span
      className="loader_dot"
    />
    <span
      className="loader_dot"
    />
    <span
      className="loader_dot"
    />
  </span>
</button>
`;

exports[`Button Component it renders caution 1`] = `
<button
  className=" btn__caution "
  type="button"
/>
`;

exports[`Button Component it renders children 1`] = `
<button
  className="btn "
  type="button"
>
  Button Children
</button>
`;

exports[`Button Component it renders className 1`] = `
<button
  className="btn test classnames"
  type="button"
/>
`;

exports[`Button Component it renders disabled 1`] = `
<button
  className=" btn__disabled "
  type="button"
/>
`;

exports[`Button Component it renders html tagName 1`] = `
<span
  className="btn "
  type="button"
/>
`;

exports[`Button Component it renders link 1`] = `
<button
  className=" btn__link "
  type="button"
/>
`;

exports[`Button Component it renders pimary && secondary 1`] = `
<button
  className="btn__primary"
>
  'primary' and 'secondary' are mutually exclusive.
</button>
`;

exports[`Button Component it renders primary && alt 1`] = `
<button
  className="btn__primary"
>
  'primary' and 'alt' are mutually exclusive.
</button>
`;

exports[`Button Component it renders primary 1`] = `
<button
  className="btn__primary "
  type="button"
/>
`;

exports[`Button Component it renders secondary 1`] = `
<button
  className="btn__secondary "
  type="button"
/>
`;
